### PR TITLE
Test: implement kargs e2e test

### PR DESCRIPTION
**Pull request by lukewarmtemp**
Integration test for the changes in: https://github.com/containers/bootc/pull/401

### CODE REVIEW

**Comment by lukewarmtemp** [comment]
@cgwalters @henrywang Sorry, do you guys think you'd be able to help me figure out how to stop the integration tests from skipping?

**Comment by henrywang** [comment]
Hi! I think we can reuse `image-install-upgrade.sh` and `check-system.yaml`.
```diff
diff --git i/tests/integration/image-install-upgrade.sh w/tests/integration/image-install-upgrade.sh
index 2468802..bc668bc 100755
--- i/tests/integration/image-install-upgrade.sh
+++ w/tests/integration/image-install-upgrade.sh
@@ -154,7 +154,7 @@ case "$IMAGE_TYPE" in
             -v /dev:/dev \
             -v .:/output \
             "$TEST_IMAGE_URL" \
-            bootc install to-disk --filesystem "$ROOTFS" --generic-image --via-loopback /output/disk.raw
+            bootc install to-disk --filesystem "$ROOTFS" --generic-image --via-loopback --karg=nosmt /output/disk.raw
 
         sudo qemu-img convert -f raw ./disk.raw -O qcow2 "/var/lib/libvirt/images/disk.qcow2"
         rm -f disk.raw
@@ -194,6 +194,7 @@ ansible-playbook -v \
     -e test_os="$TEST_OS" \
     -e bootc_image="$TEST_IMAGE_URL" \
     -e image_label_version_id="$REDHAT_VERSION_ID" \
+    -e karg="nosmt" \
     playbooks/check-system.yaml
 
 greenprint "Create upgrade Containerfile"
diff --git i/tests/integration/playbooks/check-system.yaml w/tests/integration/playbooks/check-system.yaml
index ede5a3a..6a9bae1 100644
--- i/tests/integration/playbooks/check-system.yaml
+++ w/tests/integration/playbooks/check-system.yaml
@@ -4,6 +4,7 @@
   vars:
     bootc_image: ""
     upgrade: ""
+    karg: ""
     total_counter: "0"
     failed_counter: "0"
 
@@ -452,6 +453,31 @@
       when: ansible_facts['distribution_version'] != "41"
       # workaround for issue https://gitlab.com/fedora/bootc/base-images/-/issues/10
 
+    # case: check karg
+    - name: grep for kargs
+      shell: cat /proc/cmdline | grep {{ karg }}
+      register: kargs_check
+      # grep will exit with 1 when no results found.
+      # This causes the task not to halt play.
+      ignore_errors: true
+      when: karg != ''
+
+    - name: check if kargs exist
+      block:
+        - assert:
+            that:
+              - kargs_check is succeeded
+            fail_msg: install kargs not found
+            success_msg: install kargs found
+      always:
+        - set_fact:
+            total_counter: "{{ total_counter | int + 1 }}"
+      rescue:
+        - name: failed count + 1
+          set_fact:
+            failed_counter: "{{ failed_counter | int + 1 }}"
+      when: karg != ''
+
     # case: check reboot times
     - name: check reboot times
       command: last reboot
```

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -58,8 +58,8 @@ jobs:
       repo_url: ${{ fromJson(steps.pr-api.outputs.data).head.repo.html_url }}
 
   integration:
-    needs: pr-info
-    if: ${{ needs.pr-info.outputs.allowed_user == 'true' && !contains(github.event.pull_request.labels.*.name, 'control/skip-ci') }}
+    # needs: pr-info
```
Comment (henrywang) [inline .github/workflows/integration.yml:61]: The workflow changes in PR will not be available in this PR.

Comment (lukewarmtemp) [reply .github/workflows/integration.yml:61]: Would I be able to get the `admin` permission? cc @cgwalters

Comment (henrywang) [reply .github/workflows/integration.yml:61]: Oh, sorry, should be `admin` or `write` permission.

Comment (cgwalters) [reply .github/workflows/integration.yml:61]: @lukewarmtemp I previously sent you an invite to the org.  Just sent you a new one.  Can you check your spam folder?

Comment (lukewarmtemp) [reply .github/workflows/integration.yml:61]: Got it, thanks!

**Comment by cgwalters** [comment]
I forgot to mention here that I added kargs tests in https://github.com/containers/bootc/pull/620/commits/b77745905267ac6edd5b34d8255115a38182bf75 (that said, it makes sense to have some in the e2e tests too)

**Review by cgwalters — APPROVED** [review]

**Pull request merged**